contracts/VestingEscrowFactory.vy

```diff
@@ -1,79 +1,149 @@
+@author Curve Finance, Yearn Finance, Lido Finance
 @license MIT
-@notice Stores and distributes ERC20 tokens by deploying `VestingEscrowSimple` contracts
+@notice Stores and distributes ERC20 tokens by deploying `VestingEscrow` or `VestingEscrowFullyRevokable` contracts
 """
 
 from vyper.interfaces import ERC20
 
 
-interface VestingEscrowSimple:
+interface IVestingEscrow:
     def initialize(
-        admin: address,
         token: address,
-        recipient: address,
         amount: uint256,
+        recipient: address,
+        owner: address,
+        manager: address,
         start_time: uint256,
         end_time: uint256,
         cliff_length: uint256,
+        voting_adapter_addr: address,
     ) -> bool: nonpayable
 
 
 event VestingEscrowCreated:
-    funder: indexed(address)
+    creator: indexed(address)
     token: indexed(address)
+    amount: uint256
     recipient: indexed(address)
+    owner: address
+    manager: address
     escrow: address
-    amount: uint256
+    escrow_type: uint256  # 0 - simple, 1 - fully revokable
     vesting_start: uint256
     vesting_duration: uint256
     cliff_length: uint256
+    voting_adapter: address
+
 
+event ERC20Recovered:
+    token: address
+    amount: uint256
+
+
+target_simple: public(address)
+target_fully_revokable: public(address)
+voting_adapter: public(address)
+owner: public(address)
 
-target: public(address)
 
 @external
-def __init__(target: address):
+def __init__(
+    target_simple: address,
+    target_fully_revokable: address,
+    voting_adapter: address,
+):
     """
     @notice Contract constructor
-    @dev Prior to deployment you must deploy one copy of `VestingEscrowSimple` which
-         is used as a library for vesting contracts deployed by this factory
-    @param target `VestingEscrowSimple` contract address
+    @dev Prior to deployment you must deploy one copy of `VestingEscrowSimple` and `VestingEscrowFullyRevokable` which
+         are used as a library for vesting contracts deployed by this factory
+    @param target_simple `VestingEscrow` contract address
+    @param target_fully_revokable `VestingEscrowFullyRevokable` contract address
+    @param voting_adapter Address of the Lido Voting Adapter
     """
-    self.target = target
+    assert target_simple != empty(
+        address
+    ), "target_simple should not be ZERO_ADDRESS"
+    assert target_fully_revokable != empty(
+        address
+    ), "target_fully_revokable should not be ZERO_ADDRESS"
+    self.target_simple = target_simple
+    self.target_fully_revokable = target_fully_revokable
+    self.voting_adapter = voting_adapter
+    self.owner = msg.sender
 
 
 @external
 def deploy_vesting_contract(
     token: address,
-    recipient: address,
     amount: uint256,
+    recipient: address,
+    owner: address,
     vesting_duration: uint256,
     vesting_start: uint256 = block.timestamp,
     cliff_length: uint256 = 0,
+    escrow_type: uint256 = 0,  # use simple escrow by default
```

Comment:
Is `uint8` or `uint16` not enough?

Reply:
It is not stored in the storage so no difference